provider/azure/doc.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package azure implements the Azure provider, registered with
+// environs under the name "azure". The provider implements the
+// [github.com/juju/juju/environs.Environ] interface, which defines
+// methods for provisioning compute, network, and storage resources.
+//
+// Here we describe some key implementation details specific to the Azure provider.
+//
+// # SDK
+//
+// The provider implementation is built using the [Azure SDK].
+//
+// # Credential Types
+//
+// The supported credential types are:
+//   - service-principal-secret
+//   - managed-identity
+//
+// The recommended way to create a credential with which to bootstrap is
+// to run
+//
+//	juju add-credential azure
+//
+// and follow the prompts. Choosing "interactive" is the best choice for
+// setting up a service principal credential as the browser is used to
+// log into an Azure account and the credential attributes are filled
+// in automatically.
+//
+// # Resource Groups
+//
```

Comment:
Change title to "Models". (Basically, think of it as "I'm talking to a competent developer who knows about environs, i.e., how we generally do things, and wants to find out how we adjust things for azure specifically.)